runtime/parachains/src/configuration.rs

```diff
@@ -258,37 +255,68 @@ impl<BlockNumber: Zero> HostConfiguration<BlockNumber> {
 	}
 }
 
-pub trait Config: frame_system::Config + shared::Config { }
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
 
-decl_storage! {
-	trait Store for Module<T: Config> as Configuration {
-		/// The active configuration for the current session.
-		ActiveConfig get(fn config) config(): HostConfiguration<T::BlockNumber>;
-		/// Pending configuration (if any) for the next session.
-		PendingConfig: map hasher(twox_64_concat) SessionIndex => Option<HostConfiguration<T::BlockNumber>>;
-	}
-	add_extra_genesis {
-		build(|config: &Self| {
-			config.config.check_consistency();
-		})
-	}
-}
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
 
-decl_error! {
-	pub enum Error for Module<T: Config> {
+	#[pallet::config]
+	pub trait Config: frame_system::Config + shared::Config {}
+
+	#[pallet::error]
+	pub enum Error<T> {
 		/// The new value for a configuration parameter is invalid.
 		InvalidNewValue,
 	}
-}
 
-decl_module! {
-	/// The parachains configuration module.
-	pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
-		type Error = Error<T>;
+	/// The active configuration for the current session.
+	#[pallet::storage]
+	#[pallet::getter(fn config)]
+	pub type ActiveConfig<T: Config> = StorageValue<
+		_,
+		HostConfiguration<T::BlockNumber>,
+		ValueQuery
+	>;
+
+	/// Pending configuration (if any) for the next session.
+	#[pallet::storage]
+	pub type PendingConfig<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		SessionIndex,
+		HostConfiguration<T::BlockNumber>
+	>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub config: HostConfiguration<T::BlockNumber>
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			GenesisConfig {
+				config: Default::default()
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			ActiveConfig::<T>::put(&self.config);
+			self.config.check_consistency();
+		}
+	}
```

Comment:
I Believe that is equivalent to the old macro... but thought it would be could to highlight

Reply:
Yes, it looks right, but you could add a unit test to confirm if you like :)

Reply:
Also the order of these lines should be swapped. First we check consistency, then we put the value into storage.

shouldn't practically matter since ultimately it results in a panic, but reading the code doesnt make sense

Reply:
ah good point doesn't do much checking before we put the config storage lol